design/BufferOperations.md

```diff
@@ -156,7 +159,7 @@ function bufferSubData(device, destBuffer, destOffset, srcArrayBuffer) {
     const byteCount = srcArrayBuffer.byteLength;
     const (srcBuffer, mapping) = device.createBufferMapped({
         size: byteCount,
-        usage: GPUBufferUsage.MAP_WRITE | GPUBufferUsage.TRANSFER_SRC,
+        usage: GPUBufferUsage.TRANSFER_SRC,
     });
     (new Uint8Array(mapping)).set(new Uint8Array(srcArrayBuffer)); // memcpy
     srcBuffer.unmap();
```

Comment:
This would fail if we keep "buffer must have been created with the MAP_READ or the MAP_WRITE usage flags." from earlier in this document. 

Assuming the first "unmap()" call doesn't fail on a buffer created this way, what about subsequent calls? Unmap()ping a mapped buffer is not supposed to be an error either.

Reply:
Also, would it then be legal to call `createBufferMapped` with `MAP_READ`, write to the ArrayBuffer, `unmap()`, and immediately call `mapReadAsync` and view those changes?

Reply:
Oops, I didn't notice https://github.com/kainino0x/gpuweb/pull/5. I'll post any add'l comments there.

Reply:
I haven't decided whether to propose that one yet.

You're right about both comments above. I will need to update this PR more to fix the state tracking rules. (marking unresolved so I can do that if I keep this proposal)

Reply:
Since we've decided to merge this, gonna resolve this thread for now.